Add some guidelines for contributors on how to add to the errors catalog (#2269)
grafana/mimir: docs/internal/contributing/README.md

Note: if you attempt to view pages on Github, it's likely that you might find broken links or pages. That is expected and should not be addressed unless it is causing issues with the site that occur as part of the build.

## Errors catalog

We document the common user-visible errors so it is easy for the user to search for how to address those errors when they see them.

To add a new error:

> - Under `pkg/util/globalerror/errors.go`, create a new unique ID string as a constant. This string should not be changed once your changes make it into a public release.

Reviewer: ```suggestion
- Under `pkg/util/globalerror/errors.go`, create a new unique ID string as a constant. After your changes make it into a public release, do not change this string.
```

> - When returning the error, use one of the functions in `globalerror` to generate the message. If you return the same error from multiple places, create a new function to return the error so the message string is defined in only one place, then add a simple test for that function comparing its actual output to the expected message as a hardcoded string.

Reviewer: ```suggestion
- When returning the error, use one of the functions in `globalerror` to generate the message. If you return the same error from multiple places, create a new function to return a unique error so that each message string is defined in one place. Then, add a simple test (as a hard-coded string) for that function to compare its actual output to the expected message.
```

Author: Modified this slightly as the expected message is the hard-coded string.
